src/core/blockchain/zilliqa/client.ts

```diff
@@ -28,17 +30,15 @@ export class Client extends BlockchainGenericClient {
 
     public async getBalance(address: string): Promise<BigNumber> {
         try {
-            const response = await this.call('GetBalance', [
-                fromBech32Address(address)
-                    .replace('0x', '')
-                    .toLowerCase()
-            ]);
-            return new BigNumber(response.result.balance);
-        } catch (result) {
-            if (result?.error?.message === 'Account is not created') {
-                return Promise.resolve(new BigNumber(0));
-            }
-            return Promise.reject(result);
+            const data = await new ApiClient().validators.getBalance(
+                address,
+                Blockchain.ZILLIQA,
+                this.chainId.toString()
+            );
+
+            return data?.balance?.total || new BigNumber(0);
```

Comment:
we should return undefined here, and where we display the balances if the balance is undefined we should display `N/A` instead of `0`. When we display 0 users got scared they lost their tokens :)

Reply:
not necessary to implement it now, since we need to do it in many places. If it's simple do it, else add it in ZenHub.